tests/cli/test_main_list.py

```diff
@@ -249,3 +249,26 @@ def test_fields_invalid(conda_cli):
     )
     assert "list_fields" in str(exc)
     assert "invalid-field" in str(exc)
+
+
+def test_exit_codes(conda_cli):
+    # If the package is installed, with or without --check, the exit code must be 0
+    out, err, rc = conda_cli("list", f"--prefix={sys.prefix}", "conda")
+    assert rc == 0
+    out, err, rc = conda_cli("list", f"--prefix={sys.prefix}", "--check", "conda")
+    assert rc == 0
+
+    # For not installed packages, the exit code will be 1 only if --check is passed
+    out, err, rc = conda_cli("list", f"--prefix={sys.prefix}", "does-not-exist")
+    assert rc == 0
+    out, err, rc = conda_cli(
+        "list", f"--prefix={sys.prefix}", "--check", "does-not-exist"
+    )
+    assert rc == 1
+
+    # Make sure it works with --json too
+    out, err, rc = conda_cli(
+        "list", f"--prefix={sys.prefix}", "--check", "--json", "does-not-exist"
```

Comment:
For completeness, do we also want to add a test case with `--json` but without `--check`?

Reply:
I don't think it's necessary, imo.